src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetRandomCommand.cs

```diff
@@ -306,23 +306,25 @@ private double GetRandomDouble(double min, double max)
             {
                 do
                 {
-                    double r = this.Generator.NextDouble();
-                    randomNumber = min + r * max - r * min;
+                    double r = Generator.NextDouble();
+                    result = min + r * max - r * min;
                 }
-                while (randomNumber >= max);
+                while (result >= max);
             }
             else
             {
                 do
                 {
-                    double r = this.Generator.NextDouble();
-                    randomNumber = min + r * diff;
+                    double r = Generator.NextDouble();
+                    result = min + r * diff;
                     diff = diff * r;
                 }
-                while (randomNumber >= max);
+                while (result >= max);
             }
 
-            return randomNumber;
+            Debug.Assert(min <= result, "lower bound <= random number");
+            Debug.Assert(result < max, "random number < upper bound");
```

Comment:
Why first condition is <= but second one <?

Reply:
GetRandomDouble returns a random number between minValue (included) and maxValue (excluded). Hence the presence of the do{}while(randomNumber >= maxValue) loop